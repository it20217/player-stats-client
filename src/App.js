import { React, useContext, useState, useEffect } from 'react';
import AuthContext from './store/auth-context';
import { Routes, Route, useLocation } from 'react-router-dom';
import './App.module.scss';
import Header from './components/Header/Header';
import Home from './views/Home/Home';
import Login from './views/Login/Login';
import UserProfile from './views/UserProfile/UserPage';
import Notification from './components/Notification/Notification';
import SignUp from './views/SignUp/SignUp';
import PlayerManagement from './views/PlayerManagement/PlayerManagement';
import PlayerProfile from './views/PlayerProfile/PlayerProfile';
import PlayerEvaluation from './views/PlayerEvaluation/PlayerEvaluation';
import RequireAuth from './components/RequireAuth/RequireAuth';
import UserManagement from './views/UserManagement/UserManagement';
import EventManagement from './views/EventManagement/EventManagement';
import Sidebar from './components/Sidebar/Sidebar';

function App() {

  const AuthCtx = useContext(AuthContext);
  const isLoggedIn = AuthCtx.isLoggedIn;
  const location = useLocation();
  const path = location.pathname;

  const [notificationOpen, setNotificationOpen] = useState(false);
  const [notification, setNotification] = useState({
    type: '',
    message: ''
  });

  useEffect(() => {
    setNotification({
      type: AuthCtx.notification?.type || 'info',
      message: AuthCtx.notification?.message || '',
      delay: AuthCtx.notification?.delay || 5000
    });
    if (AuthCtx.notification) {
      setNotificationOpen(true);
    }
  }, [AuthCtx.notification]);

  return (
    <div className='App'>
      <Header/>
      <div className='userContainer flex flex-row w-full justify-between'>
        {isLoggedIn && path !== '/' &&<Sidebar/>}
          <div className="w-full">
            <Routes>
              <Route 
                path='/' 
                element={<Home />} 
              />
              <Route 
                path='/login' 
                element={<Login />} 
              />
              <Route 
                path='/signup' 
                element={<SignUp />} 
              />
              <Route element={<RequireAuth />}>
                <Route 
                  path='/user' 
                  element={<UserProfile />} 
                />
                <Route 
                  path='/usermanagement' 
                  element={<UserManagement />} 
                />
                <Route 
                  path='/playermanagement' 
                  element={<PlayerManagement />} 
                />
                <Route 
                  path='/player/:id' 
                  element={<PlayerProfile />} 
                /> 
                <Route 
                  path='/events' 
                  element={<EventManagement />} 
                />
                <Route 
                  path='/playerevaluation' 
                  element={<PlayerEvaluation />} 
                />
                <Route 
                  path='/playerevaluation/:eventId' 
                  element={<PlayerEvaluation />} 
                />
                <Route 
                  path='/playerevaluation/:eventId/assignment/:assignmentId' 
                  element={<PlayerEvaluation />} 
                />
                <Route 
                  path='/playerevaluation' 
                  element={<PlayerEvaluation />} 
                />
              </Route>
            </Routes>
          </div>
      </div>
      {notificationOpen && 
        <Notification 
          notification={notification} 
          delay={notification.type === 'info' || notification.type === 'success' ? 5000 : null} 
          close={() => {setNotificationOpen(false); setNotification({
            type: '',
            message: ''
          })}} 
        />
      }
    </div>
  );
}

export default App;
