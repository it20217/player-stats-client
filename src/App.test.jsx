import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import App from './App';

describe("App component",() => {
  it("renders correctly", () => {
    const component = render(
      <MemoryRouter initialEntries={["/"]}>
        <App/>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
