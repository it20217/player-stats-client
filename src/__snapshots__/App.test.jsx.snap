// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="Header"
        >
          <div
            class="img-container"
          >
            <img
              alt="Logo"
              src="Logo.png"
            />
          </div>
          <div
            class="text-logo"
          >
            <h1
              class="text-2xl font-bold text-gray-900 pl-7"
            >
              Football
              <span
                class="text-2xl font-normal italic"
              >
                 academy
              </span>
            </h1>
          </div>
          <div
            class="btn-container"
          >
            <a
              class="mx-2 my-2 bg-white transition duration-150 ease-in-out hover:bg-slate-700 hover:border-slate-800 hover:text-yellow-400 rounded border border-slate-800 text-gray-800 px-10 py-4 text-sm hover:bg-slate-100 focus:outline-none focus:ring-2 focus:ring-offset-2  focus:ring-slate-100"
              href="/login"
            >
              Login
            </a>
          </div>
        </div>
        <div
          class="userContainer flex flex-row w-full justify-between"
        >
          <div
            class="w-full"
          >
            <div
              class="overflow-y-hidden"
            >
              <div
                class="relative flex justify-center items-center md:flex md:justify-end"
              >
                <img
                  alt="santa"
                  class="soccer-player-image"
                  src="soccer-player.webp"
                />
                <div
                  class="w-full flex absolute justify-start flex-col items-center md:py-10 lg:py-20 xl:py-40 md:space-y-12 md:space-y-0"
                >
                  <div
                    class=" py-40 sm:py-20  md:hidden"
                  />
                  <div
                    class="w-full custom sm:mt-96 md:mt-0 h-full flex lg:h-96 px-4 z-10 items-end md:items-end md:px-6 xl:px-20 whitespace-nowrap"
                  >
                    <h1
                      class="text-6xl font-bold text-gray-900 mr-6 "
                    >
                      Player 
                      <span
                        class="text-6xl font-normal"
                      >
                         perfomance
                      </span>
                    </h1>
                    <hr
                      class="w-2.5 h-16 bg-slate-700"
                    />
                    <h1
                      class="text-6xl font-bold text-gray-900 pl-7"
                    >
                      Talent
                      <span
                        class="text-6xl font-normal"
                      >
                         scouting
                      </span>
                    </h1>
                  </div>
                  <div
                    class="w-full custom h-full flex lg:h-96 px-4 z-10 items-start md:items-start md:px-6 xl:px-20 "
                  >
                    <button
                      class="mt-6 w-full md:w-auto lg:mt-8 py-2 md:py-3 px-10 flex justify-center duration-700 hover:-translate-y-1 ease-in-out items-center relative rounded-md text-lg font-medium text-white"
                    >
                      <p
                        class="relative z-10"
                      >
                        Explore Now
                      </p>
                      <img
                        alt="gradient"
                        class="absolute hidden md:block w-full h-full z-0"
                        src="https://tuk-cdn.s3.amazonaws.com/can-uploader/hero_8_btn.png"
                      />
                      <img
                        alt="gradient"
                        class="md:hidden w-full h-full z-0 absolute"
                        src="https://tuk-cdn.s3.amazonaws.com/can-uploader/hero_8-btn_mobile.png"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="Header"
      >
        <div
          class="img-container"
        >
          <img
            alt="Logo"
            src="Logo.png"
          />
        </div>
        <div
          class="text-logo"
        >
          <h1
            class="text-2xl font-bold text-gray-900 pl-7"
          >
            Football
            <span
              class="text-2xl font-normal italic"
            >
               academy
            </span>
          </h1>
        </div>
        <div
          class="btn-container"
        >
          <a
            class="mx-2 my-2 bg-white transition duration-150 ease-in-out hover:bg-slate-700 hover:border-slate-800 hover:text-yellow-400 rounded border border-slate-800 text-gray-800 px-10 py-4 text-sm hover:bg-slate-100 focus:outline-none focus:ring-2 focus:ring-offset-2  focus:ring-slate-100"
            href="/login"
          >
            Login
          </a>
        </div>
      </div>
      <div
        class="userContainer flex flex-row w-full justify-between"
      >
        <div
          class="w-full"
        >
          <div
            class="overflow-y-hidden"
          >
            <div
              class="relative flex justify-center items-center md:flex md:justify-end"
            >
              <img
                alt="santa"
                class="soccer-player-image"
                src="soccer-player.webp"
              />
              <div
                class="w-full flex absolute justify-start flex-col items-center md:py-10 lg:py-20 xl:py-40 md:space-y-12 md:space-y-0"
              >
                <div
                  class=" py-40 sm:py-20  md:hidden"
                />
                <div
                  class="w-full custom sm:mt-96 md:mt-0 h-full flex lg:h-96 px-4 z-10 items-end md:items-end md:px-6 xl:px-20 whitespace-nowrap"
                >
                  <h1
                    class="text-6xl font-bold text-gray-900 mr-6 "
                  >
                    Player 
                    <span
                      class="text-6xl font-normal"
                    >
                       perfomance
                    </span>
                  </h1>
                  <hr
                    class="w-2.5 h-16 bg-slate-700"
                  />
                  <h1
                    class="text-6xl font-bold text-gray-900 pl-7"
                  >
                    Talent
                    <span
                      class="text-6xl font-normal"
                    >
                       scouting
                    </span>
                  </h1>
                </div>
                <div
                  class="w-full custom h-full flex lg:h-96 px-4 z-10 items-start md:items-start md:px-6 xl:px-20 "
                >
                  <button
                    class="mt-6 w-full md:w-auto lg:mt-8 py-2 md:py-3 px-10 flex justify-center duration-700 hover:-translate-y-1 ease-in-out items-center relative rounded-md text-lg font-medium text-white"
                  >
                    <p
                      class="relative z-10"
                    >
                      Explore Now
                    </p>
                    <img
                      alt="gradient"
                      class="absolute hidden md:block w-full h-full z-0"
                      src="https://tuk-cdn.s3.amazonaws.com/can-uploader/hero_8_btn.png"
                    />
                    <img
                      alt="gradient"
                      class="md:hidden w-full h-full z-0 absolute"
                      src="https://tuk-cdn.s3.amazonaws.com/can-uploader/hero_8-btn_mobile.png"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
